OsmAnd/src/net/osmand/plus/myplaces/FavouritesHelper.java

```diff
@@ -602,7 +603,7 @@ public void sortAll() {
 		for (FavoriteGroup g : favoriteGroups) {
 			Collections.sort(g.getPoints(), favoritesComparator);
 		}
-		Collections.sort(cachedFavoritePoints, favoritesComparator);
```

Comment:
Collections should be always recreated (!) in case they are changed to avoid all possible modify exceptions if they accessed from multiple threads.